src/components/PopupWithForm.js

```diff
@@ -3,35 +3,70 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super(popupSelector);
-    this._modalForm = this._modalElement.querySelector(".modal__form");
+    this._popupForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
-    this._inputList = this._modalForm.querySelectorAll(".modal__input");
-  }
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+    this._inputList = this._popupForm.querySelectorAll(".modal__input");
 
-  getForm() {
-    return this._modalForm
+    console.log("Formulario seleccionado:", this._popupForm); // Verifica si se selecciona el formulario
+    console.log("Botón de submit seleccionado:", this._submitButton); // Verifica si el botón es seleccionado
   }
 
   _getInputValues() {
-    this._inputData = {};
+    this._formValues = {};
     this._inputList.forEach((input) => {
-      this._inputData[input.name] = input.value;
+      this._formValues[input.name] = input.value;
     });
-    return this._inputData;
+    return this._formValues;
+  }
+
+  // Función para habilitar o deshabilitar el botón de envío
+  _toggleSubmitButtonState() {
+    const isFormValid = this._popupForm.checkValidity();
+    if (isFormValid) {
+      this._submitButton.removeAttribute("disabled");
+      this._submitButton.classList.remove("modal__button_disabled");
+    } else {
+      this._submitButton.setAttribute("disabled", "true");
+      this._submitButton.classList.add("modal__button_disabled");
```

Comment:
Please, delete `_toggleSubmitButtonState` from `PopupWithForm`

Now only `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it). It’s a violation of **Single responsibility** principle